test/gas/gas.js

```diff
@@ -0,0 +1,563 @@
+      const options = {
+        blockTime: 0.5, // seconds
+        mnemonic
+      };
+      const { accounts, web3 } = await getWeb3(options);
+
+      const transactions = [
+        {
+          value: "0x10000000",
+          gasLimit: "0x33450",
+          from: accounts[2],
+          to: accounts[1],
+          nonce: "0x0"
+        },
+        {
+          value: "0x10000000",
+          gasLimit: "0x33450",
+          from: accounts[2],
+          to: accounts[1],
+          nonce: "0x1"
+        },
+        {
+          value: "0x10000000",
+          gasLimit: "0x33450",
+          from: accounts[1], // <
+          to: accounts[2], // <^ reversed tx order
+          nonce: "0x0"
+        }
+      ];
+
+      // Precondition
+      const initialBlockNumber = await web3.eth.getBlockNumber();
+      assert.deepStrictEqual(initialBlockNumber, 0, "Current Block Should be 0");
+
+      const localGasInstance = await deployContract(abi, accounts, bytecode, web3);
+
+      // prime storage by making sure it is set to 0
+      await localGasInstance.methods.reset().send({ from: accounts[0], gas: 5000000 });
+      const method = localGasInstance.methods.triggerAllRefunds();
+      const gasEstimate = await method.estimateGas({ from: accounts[0] });
+
+      const hashes = await Promise.all(
+        transactions.map((transaction) => {
+          const promiEvent = web3.eth.sendTransaction(transaction);
+
+          return new Promise((resolve) => {
+            promiEvent.once("transactionHash", async(hash) => {
+              // Ensure there's no receipt since the transaction hasn't yet been processed. Ensure IntervalMining
+              const receipt = await web3.eth.getTransactionReceipt(hash);
+              assert.strictEqual(receipt, null, "No receipt since the transaction hasn't yet been processed.");
+              resolve(hash);
+            });
+          });
+        })
+      );
+
+      const currentBlockNumber = await web3.eth.getBlockNumber();
+      assert.deepStrictEqual(currentBlockNumber, 2, "Current Block Should be 2");
+
+      const { gasUsed } = await method.send({ from: accounts[0], gas: gasEstimate });
+
+      let transactionCostMinusRefund = gasEstimate - RSELFDESTRUCT_REFUND - RSCLEAR_REFUND;
+      if (provider.options.hardfork === "byzantium") {
+        assert.strictEqual(gasUsed, transactionCostMinusRefund);
+      } else if (provider.options.hardfork === "constantinople") {
+        // since storage was initially primed to 0 and we call triggerAllRefunds(), which then
+        // resets storage back to 0, 19800 gas is added to the refund counter per Constantinople EIP 1283
+        transactionCostMinusRefund =
+          gasEstimate - RSELFDESTRUCT_REFUND - RSCLEAR_REFUND_FOR_RESETTING_DIRTY_SLOT_TO_ZERO;
+        assert.strictEqual(gasUsed, transactionCostMinusRefund);
+      }
+
+      const [tx1, tx2, tx3] = await Promise.all(hashes.map((hash) => web3.eth.getTransactionReceipt(hash)));
```

Comment:
Two things here:

This used to be an array called `receipts` - why change to `tx`? Often we use `tx` to mean "parameters for a transaction we're about to run" - sometimes it also means "the hash of the transaction we just ran" or "the result of `eth_getTransactionBy*` - let's avoid overloading the term further.

This is definitely a case where I would've preferred not to destructure. Any time you have a bunch of variables named the same thing with but differentiated by a numeric suffix, you should have an array. Keeping them as an array makes it easier to do bulk FP-style operations.

Reply:
Can change this back to a receipts array.

Reply:
We have `receipts` again.